Sources/Testing/Running/Configuration.TestFilter.swift

```diff
@@ -280,12 +343,23 @@ extension Configuration.TestFilter {
   }
 }
 
+extension Configuration.TestFilter {
+  /// A type describing an error encountered while performing test filtering.
+  fileprivate enum Error: Swift.Error {
```

Comment:
I recently added `SystemError`, can we use that? Or, perhaps this is a reason to coalesce our several error types into a single `TestingError` with a few different cases.